### PR TITLE
Extend unary executable operators

### PR DESCRIPTION
**Pull request by chengsijin0817**
Extend three executable operators to make them applicable to SPARQL query patterns

### CODE REVIEW

File: src/main/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpGenericBindJoinWithRequestOps.java [hunk 1]
```diff
@@ -78,4 +87,28 @@ public void send( final SolutionMapping smFromRequest ) {
...
+	 */
+	protected Op representQueryPatternAsJenaOp( final QueryType query ) {
+		if ( query instanceof SPARQLGraphPattern ) {
+			if ( query instanceof TriplePattern) {
+				return new OpTriple( ((TriplePattern)query).asJenaTriple());
+			}
+			else if (query instanceof BGP) {
+				final Set<TriplePattern> tps = (Set<TriplePattern>) ((BGP) query).getTriplePatterns();
+				final BasicPattern bgp = new BasicPattern();
+				for (TriplePattern tp : tps) {
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpGenericBindJoinWithRequestOps.java:103]:
```suggestion
				final BasicPattern bgp = new BasicPattern();
				for ( final TriplePattern tp : ((BGP) query).getTriplePatterns() ) {
```

File: src/main/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpBindJoinSPARQLwithFILTER.java [hunk 2]
```diff
@@ -21,12 +20,13 @@
...
+import se.liu.ida.hefquin.engine.query.BGP;
 import se.liu.ida.hefquin.engine.query.SPARQLGraphPattern;
 import se.liu.ida.hefquin.engine.query.TriplePattern;
 import se.liu.ida.hefquin.engine.query.impl.QueryPatternUtils;
 import se.liu.ida.hefquin.engine.query.impl.SPARQLGraphPatternImpl;
 
-public class ExecOpBindJoinSPARQLwithFILTER extends ExecOpGenericBindJoinWithRequestOps<TriplePattern,SPARQLEndpoint>
+public class ExecOpBindJoinSPARQLwithFILTER extends ExecOpGenericBindJoinWithRequestOps<SPARQLGraphPattern, SPARQLEndpoint>
 {
 	protected final Set<Var> varsInTP;
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpBindJoinSPARQLwithFILTER.java:31]: Please rename this member variable to `varsInSubQuery`.

File: src/main/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpBindJoinSPARQLwithUNION.java [hunk 3]
```diff
@@ -20,12 +19,13 @@
...
+import se.liu.ida.hefquin.engine.query.BGP;
 import se.liu.ida.hefquin.engine.query.SPARQLGraphPattern;
 import se.liu.ida.hefquin.engine.query.TriplePattern;
 import se.liu.ida.hefquin.engine.query.impl.QueryPatternUtils;
 import se.liu.ida.hefquin.engine.query.impl.SPARQLGraphPatternImpl;
 
-public class ExecOpBindJoinSPARQLwithUNION extends ExecOpGenericBindJoinWithRequestOps<TriplePattern,SPARQLEndpoint>
+public class ExecOpBindJoinSPARQLwithUNION extends ExecOpGenericBindJoinWithRequestOps<SPARQLGraphPattern, SPARQLEndpoint>
 {
 	protected final List<Var> varsInTP;
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpBindJoinSPARQLwithUNION.java:30]: Please rename this member variable to `varsInSubQuery`.

File: src/main/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpBindJoinSPARQLwithVALUES.java [hunk 4]
```diff
@@ -19,12 +18,13 @@
...
+import se.liu.ida.hefquin.engine.query.BGP;
 import se.liu.ida.hefquin.engine.query.SPARQLGraphPattern;
 import se.liu.ida.hefquin.engine.query.TriplePattern;
 import se.liu.ida.hefquin.engine.query.impl.QueryPatternUtils;
 import se.liu.ida.hefquin.engine.query.impl.SPARQLGraphPatternImpl;
 
-public class ExecOpBindJoinSPARQLwithVALUES extends ExecOpGenericBindJoinWithRequestOps<TriplePattern,SPARQLEndpoint>
+public class ExecOpBindJoinSPARQLwithVALUES extends ExecOpGenericBindJoinWithRequestOps<SPARQLGraphPattern, SPARQLEndpoint>
 {
 	protected final List<Var> varsInTP;
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpBindJoinSPARQLwithVALUES.java:29]: Please rename this member variable to `varsInSubQuery`.

**Review by hartig — CHANGES_REQUESTED** [review]
I have just a few minor things. Other than these, this PR looks good from my perspective. (nonetheless, @scferrada can you please take a look as well)

**Review by hartig — APPROVED** [review]

**Review by scferrada — COMMENTED** [review]
Hi, the changes look great. Thanks!

**Pull request merged**